src/gwt/src/org/rstudio/studio/client/workbench/views/environment/dataimport/DataImport.java

```diff
@@ -63,4 +104,27 @@ public DataImportOptions getOptions()
       
       return options;
    }
+   
+   private void refreshPreview()
+   {
+      server_.previewDataImport(fileChooserTextBox_.getTextBoxText(), maxRows_, new ServerRequestCallback<JsObject>()
```

Comment:
Anytime we make an RPC call which might take a second or two we usually introduce dialog-local progress. Subclasses of ModalDialog can call getProgressIndicator to get a hold of this. You could then pass the ProgressIndicator to your DataImport widget when you create it.